src/chat/chatAgentRequestHandler.ts

```diff
@@ -0,0 +1,633 @@
+            toolName: tool.functionName,
+            toolDescription: tool.functionDescription,
+            correlationId: correlationId,
+        });
+
+        return { sqlTool, sqlToolParameters };
+    }
+
+    function handleLanguageModelError(
+        err: vscode.LanguageModelError,
+        stream: vscode.ChatResponseStream,
+        correlationId: string,
+    ): void {
+        console.error("Language Model Error:", err.message, "Code:", err.code);
+
+        const errorMessages: Record<string, string> = {
+            model_not_found:
+                "The requested model could not be found. Please check model availability or try a different model.",
+            no_permission:
+                "Access denied. Please ensure you have the necessary permissions to use this tool or model.",
+            quote_limit_exceeded:
+                "Usage limits exceeded. Try again later, or consider optimizing your requests.",
+            off_topic: "I'm sorry, I can only assist with SQL-related questions.",
+        };
+
+        const errorMessage =
+            errorMessages[err.code] ||
+            "An unexpected error occurred with the language model. Please try again.";
+
+        sendActionEvent(TelemetryViews.MssqlCopilot, TelemetryActions.Error, {
+            errorCode: err.code || "Unknown",
+            errorName: err.name || "Unknown",
+            errorMessage: errorMessage,
+            originalErrorMessage: err.message || "",
+            correlationId: correlationId,
+        });
+
+        stream.markdown(errorMessage);
+    }
+
+    async function sendToDefaultLanguageModel(
+        prompt: string,
+        model: vscode.LanguageModelChat,
+        stream: vscode.ChatResponseStream,
+        token: vscode.CancellationToken,
+    ): Promise<void> {
+        try {
+            stream.progress(`Using ${model.name} to process your request...`);
+
+            const messages = [vscode.LanguageModelChatMessage.User(prompt.trim())];
+            const options: vscode.LanguageModelChatRequestOptions = {
+                justification: "Fallback to default language model from MSSQL agent.",
+                tools: [], // No tools involved for this fallback
+            };
+
+            const chatResponse = await model.sendRequest(messages, options, token);
+
+            let replyText = "";
+            for await (const part of chatResponse.stream) {
+                if (part instanceof vscode.LanguageModelTextPart) {
+                    replyText += part.value;
+                }
+            }
+
+            if (replyText) {
+                stream.markdown(replyText);
+            } else {
+                stream.markdown("The language model did not return any output.");
+            }
+        } catch (err) {
+            console.error("Error in fallback language model call:", err);
+            stream.markdown("An error occurred while processing your request.");
```

Comment:
Do we care about localization at this point?

Reply:
No, not at this point.  We can revisit if desired.